### PR TITLE
TPC: improving simulation of distortions in MC

### PR DESCRIPTION
**Pull request by matthias-kleiner**
- adding option to recalculate (merging) the distortions in case of large scaling
- accessing the distortions of the derivative map at the distorted position for better consistency with corrections
- optimizing default parameters for calculating distortions/corrections

### CODE REVIEW

File: Detectors/TPC/simulation/src/Digitizer.cxx [hunk 1]
```diff
@@ -237,3 +237,33 @@ void Digitizer::setMeanLumiDistortionsDerivative(float meanLumi)
 {
   mSpaceChargeDer->setMeanLumi(meanLumi);
 }
+
+void Digitizer::recalculateDistortions()
+{
+  if (!mSpaceCharge || !mSpaceCharge) {
```
Comment (wiechula) [inline Detectors/TPC/simulation/src/Digitizer.cxx]:
```suggestion
  if (!mSpaceCharge || !mSpaceChargeDer) {
```

File: Detectors/TPC/spacecharge/src/SpaceCharge.cxx [hunk 2]
```diff
@@ -697,12 +697,59 @@ void SpaceCharge<DataT>::calcEField(const Side side)
...
+    initContainer(mGlobalDistdR[side], true);
+    initContainer(mGlobalDistdZ[side], true);
+    initContainer(mGlobalDistdRPhi[side], true);
+  } else {
+    initContainer(mGlobalCorrdR[side], true);
+    initContainer(mGlobalCorrdZ[side], true);
+    initContainer(mGlobalCorrdRPhi[side], true);
+  }
+
+  const auto& scScale = (type == Type::Distortions) ? scSCale->mInterpolatorGlobalCorr[side] : scSCale->mInterpolatorGlobalDist[side];
```
Comment (wiechula) [inline Detectors/TPC/spacecharge/src/SpaceCharge.cxx]: I find using two so similar variable names `scScale` and `scSCale` a bit dangerous. Perhaps you can find a better naming for one of them.

**Review by wiechula — CHANGES_REQUESTED** [review]

**Comment by wiechula** [comment]
+async-label async-2023-pbpb-apass4

**Comment by matthias-kleiner** [comment]
Thanks for the comments @wiechula . I changed the code accordingly

**Review by wiechula — APPROVED** [review]
Thanks @matthias-kleiner !

**Comment by wiechula** [comment]
@shahor02 , can you please check and approve.

**Review by shahor02 — APPROVED** [review]

**Comment by chiarazampolli** [comment]
Label removed, PR is in async-v1-01-16.

**Pull request merged**